cEP-0024.md

```diff
@@ -0,0 +1,252 @@
+that user completes at least one low difficulty issue. Even with current assign
+system, manual intervention by maintainers is needed to add user to developers
+team.
+
+## New Features
+
+### Auto-assign issue when it is mentioned in any of PR commits
+
+When a PR is opened or updated, if any of PR's commit message referenced
+any issues with keywords `Fixes` or `Closes` and there is no
+CI failure in PR, then auto assign PR author to the mentioned unassigned
+issues.
+
+If someone else is working on the mentioned issue, comment will be added
+on the PR (configurable).
+
+If PR author is not a member of the organization, then he/she can't be assigned
+to any of the issues, a label(configurable) will be added on the PR so that
+maintainers can identify these type of PRs and a comment will be added on PR
+informing PR author to get invited to the organization before working on any
+issue.
+
+coala can configure this plugin as:
+
+**Comment message when someone else is assigned to mentioned issues(use
+`{issues}` as placeholder to get links of all assigned issues):**
+`Sorry, but the issues {issues} are assigned to someone else. Please try working
+on some other issues.`
+
+**Label to represent that PR author is not a member of the org:**
+`author/non-member`
+
+### Auto-reject PR opened on certain branch names
+
+Normally, contributors should make changes in different branch from master
+branch of the forked repository, and submit a PR. But, often coala receives PRs
+from master branch of the forked repository.
+
+coala needs mechanism such that when a PR is opened or updated, GitMate
+automatically closes PRs opened on certain branch names(master), adds certain
+label to the PR so that maintainers can identify these type of PRs and comments
+on the PR informing the PR author to work on a different branch. Comment
+message, branch names, label will be configurable.
+
+coala can configure this plugin as:
+
+**Branch names:** `master`
+
+**Label:** `status/rejected`
+
+**Comment message:** `This PR is rejected. PR should never be opened in master
+branch, work on different branch other than master and open a new PR, follow
+[tutorial](http://api.coala.io/en/latest/Developers/Newcomers_Guide.html#step-
+3-creating-a-fork-and-testing-your-changes) if
+you repeat this again, you may get banned from coala.`
+
+### Auto-mark PR with necessary labels as per the reviews done
+
+When a reviewer reviews a PR, he/she either approves changes or request changes
+and leaves it there, but the PR would be still in the review queue. So, coala
+needs better mechanism to clear review queue when anyone reviews PR or when
+there is any CI failure. When a PR is opened or updated or when anyone reviews
+the PR, GitMate will automatically mark labels, as per the following
+conditions:
+
+- WIP label conditions:
+  1. Reviewer requests changes
+  2. GitMate's coala plugin finds defects
+  3. CI tests fail
+- Approved label conditions:
+  1. Reviewer approves changes
```

Comment:
Currently, `approved` label is marked only when a maintainer approves the PR. Is the organization happy with `approved` even when a developer/newcomer approves the changes?

Reply:
No, I am not adding any label if developer/newcomer is requesting/approving changes. Currently, only maintainer reviews are considered.

Reply:
This is not written in the cEP.  It *must* be described, otherwise we dont really know what this will do.

Are you only considering maintainer reviews for 'approved', or also for 'WIP' ?

If it only considers maintainer reviews for WIP , that means maintainers still need to go through the queue and *manually* label WIP any time a developer has requested changes.  That is one of the biggest workflow burdens on coala maintainers atm, and it disempowers developers.

Reply:
To consider developer reviews, we need to use teams, using teams in GitMate is not a part of my project. GitMate will definitely consider using teams in the future.

Reply:
You still have not edited this section to make that clear to the reader.